### PR TITLE
chore: rearrange some code to prepare for more span pointers

### PR DESCRIPTION
**Pull request by apiarian-datadog**
## Motivation

We're going to be adding more span pointers tests so paving the way by rearranging some of the code.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Comment by apiarian-datadog** [comment]
since there has been some flakiness with the other span pointers tests (may be due to some interaction between moto and async fastapi code) we're removing all but the flask span pointers webhost implementations. we're testing the boto3 tracing interactions, not the webserver interactions, so this should be fine. c.c. @cbeauchesne

**Comment by apiarian-datadog** [comment]
the remaining failures seem to be unrelated to my changes?

File: utils/build/docker/python/python3.12.base.Dockerfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ RUN python --version && curl --version
 # install python deps
 ENV PIP_ROOT_USER_ACTION=ignore
 RUN pip install --upgrade pip
-RUN pip install django pycryptodome gunicorn gevent requests boto3==1.34.141 'moto[ec2,s3,all]'==5.0.14
```
Comment (cbeauchesne) [inline utils/build/docker/python/python3.12.base.Dockerfile:12]: I would advocate for keeping this one : it'll check python 3.12, where the other one check python3.11

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: utils/build/docker/python/python3.12.base.Dockerfile (hunk 1 above)
Comment (apiarian-datadog) [reply utils/build/docker/python/python3.12.base.Dockerfile:12]: ah, interesting. ok, i'll rearrange things

Comment (apiarian-datadog) [reply utils/build/docker/python/python3.12.base.Dockerfile]: ok, they're back, i think!

**Review by cbeauchesne — APPROVED** [review]

**Review by christophe-papazian — APPROVED** [review]

**Pull request merged**